price-feeder/oracle/provider/binance.go

```diff
@@ -358,6 +359,30 @@ func (p *BinanceProvider) subscribePairs(pairs ...string) error {
 	return p.wsClient.WriteJSON(subsMsg)
 }
 
+// GetAvailablePairs return all available pairs symbol to susbscribe.
+func (p *BinanceProvider) GetAvailablePairs() (map[string]struct{}, error) {
+	resp, err := http.Get("https://api1.binance.com/api/v3/ticker/price")
```

Comment:
This assumes that the available tickers == available candles, is this always the case for these providers?

Reply:
I will take a look, In my idea when we subscribe to new pairs with `SubscribeCurrencyPairs` we try to subscribe to both `tickers` and `candles`

The method `GetAvailablePairs` is to be able to filter symbols to avoid keep subscribing to Wrong pairs at peggo

Reply:
👍 maybe if there's an endpoint that checks available candles we can use that in tandem with this one? 

They probably have the same exchange pairs anyways tbh, but just in case for some reason an API supports tickers and not candles for an exchange rate, might be worth it

Reply:
The binance one we could only get the [24hr Ticker Price Change Statistics](https://api1.binance.com/api/v3/ticker/24hr)
The other candle endpoints need symbol as parameter

The Gate.io also requires symbol as a parameter - https://www.gate.io/docs/apiv4/en/#market-candlesticks
The Huobi also requires symbol as a parameter - https://huobiapi.github.io/docs/spot/v1/en/#market-candlestick
Kraken we could grab the available candles with [Recent Trades](https://docs.kraken.com/rest/#operation/getRecentTrades), but it will only return the last 1000 trades...
The Okx also requires the symbol `InstId` as a parameter - https://www.okx.com/docs-v5/en/#rest-api-market-data-get-candlesticks

if you think/find another way of retrieving it, let me know ^^

Reply:
This is fine the way it is then, yeah those seem prone to issues (unless peggo knows the pairs that it needs and could query for them specifically using these methods, but I'm assuming you just need a list of what's available)